### ISSUE TITLE
PropertyNotSetInConstructor is issued although ignored

### ISSUE THREAD

**Issue by JoshuaBehrens** [vimeo/psalm#3424]
Having following situation:
```php
<?php

class A
{
    protected $foobar;
}

class B extends A
{
    public function __construct()
    {
    }
}
```

That issues:
```
ERROR: PropertyNotSetInConstructor - Property B::$foobar is not defined in constructor of B and in any methods called in the constructor (see https://psalm.dev/074)
class B extends A
```
So I start to ignore it for this class like this:
```xml
<?xml version="1.0"?>
<psalm
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xmlns="https://getpsalm.org/schema/config"
    xsi:schemaLocation="https://getpsalm.org/schema/config vendor/vimeo/psalm/config.xsd"
>
    <issueHandlers>
        <PropertyNotSetInConstructor>
            <errorLevel type="info">
                <referencedProperty name="B::$foobar"/>
            </errorLevel>
        </PropertyNotSetInConstructor>
    </issueHandlers>
</psalm>
```
But it keeps failing as I have to use the referenced property `A::$foobar` to ignore it. But this will ignore it for every class extending A but maybe just B can be excused for not initializing it.

My expectation is that the error message states the property I have to ignore and that I do not have to excuse the base class as others might not know how to handle that property right.

**Comment by JoshuaBehrens** [comment]
Yes. https://psalm.dev/r/ea83c10cc2 in my case A is from an external dependency and therefore I don't get a message for this and I can't change it.

**Comment by JoshuaBehrens** [comment]
Fun fact: in case of a property from a trait one has to refer to the trait which sort of get merged into class. Which is super weird as you can't even check whether a class is instanceof a trait.

**Comment by weirdan** [comment]
In your case it might be easier to suppress with a docblock: https://psalm.dev/r/c1048bc79d

**Comment by JoshuaBehrens** [comment]
Is there a way to refer to a specific property to ignore using the docblock?

**Comment by weirdan** [comment]
Unfortunately no.

**Comment by marcguyer** [comment]
> Fun fact: in case of a property from a trait one has to refer to the trait which sort of get merged into class. Which is super weird as you can't even check whether a class is instanceof a trait.
> 
> https://psalm.dev/r/906211b362

Confirmed that there is only 1 way to suppress `PropertyNotSetInConstructor`: Using config in `psalm.xml` and the reference must be to the trait, not the class using the trait. Note that the class-level docblock *does not* suppress the error.

**Comment by SCIF** [comment]
Probably, this bug can be closed as one property's issue can be easily suppressed: https://psalm.dev/r/a8e309e0a9

@marcguyer , @JoshuaBehrens , wdyt?
